x-pack/plugins/lens/public/datatable_visualization/utils.ts

```diff
@@ -0,0 +1,22 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import { Datatable } from 'src/plugins/expressions/public';
+import { getOriginalId } from './transpose_helpers';
+
+export function isNumericField(currentData: Datatable | undefined, accessor: string) {
+  const isNumeric =
+    currentData?.columns.find((col) => col.id === accessor || getOriginalId(col.id) === accessor)
+      ?.meta.type === 'number';
+
+  // check for array values
+  const hasFieldNumericValues =
+    isNumeric &&
+    currentData?.rows.every(({ [accessor]: value }) => typeof value === 'number' || value == null);
+
+  return { isNumeric, hasAllNumericValues: hasFieldNumericValues };
```

Comment:
Nit: This is causing the summary row to disappear completely from table and config flyout if there is an array value (which can happen because of last value). Did this happen on purpose? Or should we treat each array value as a separate value in terms of summary calculation?

Reply:
Maybe this function could workout whether numeric arrays are available as well and report that.
I planned to use this function also for other feature checks as well where non-numeric columns may lead to issues.

On the other hand, considering numeric arrays as valid numeric values may be a good solution. In this case I'd probably show a warning about the presence of array values and compute them as valid values.